src/map/clif.c

```diff
@@ -13235,7 +13235,8 @@ static void clif_parse_WeaponRefine(int fd, struct map_session_data *sd)
 
 	if (sd->menuskill_id != WS_WEAPONREFINE) //Packet exploit?
 		return;
-	if (pc_istrading(sd) || pc_isdead(sd) || pc_isvending(sd)) {
+	if ((sd->npc_id == 0 && pc_istrading(sd)) || (sd->npc_id != 0 && sd->state.using_megaphone == 0)
```

Comment:
this kind of conditions ignoring trading check at all if ``sd->npc_id != 0``. probably this is error?

Reply:
`#define pc_istrading(sd)      ( (sd)->npc_id || (sd)->state.vending || (sd)->state.buyingstore || (sd)->state.trading )`
If `sd->npc_id` is `0` all the other states aren't checked at all. So checking `pc_istrading(sd)` only if it's not `0` is valid here.